docs/dev/plugins.md

```diff
@@ -378,11 +381,105 @@ end
 
 #### Implement render()
 
-The primary responsibility you must fulfill is to implement render. Typically, you will examine the `run_data` Hash, which is provided as an accessor. Call `output(String, newline_wanted = true)` to send output.
+The primary responsibility you must fulfill is to implement render. Typically, you will examine the `run_data` structure (documented below), which is provided as an accessor. Call `output(String, newline_wanted = true)` to send output.
+
+#### Implement self.run_data_schema_constraints
+
+The run_data API is versioned. Your plugin should declare which version(s) it is compatible with by implementing a simple class method, which should return a GemSpec-like constraint. Currently, a simple warning is issued if a version mismatch occurs.
+
+The run_data version scheme follows a data-oriented SemVer approach:
+ * bump patch: fixing a bug in the provenance or description of a data element, no key changes
+ * bump minor: adding new data elements
+ * bump major: deleting or renaming data elements
+
+Prior to version 1.0.0, the API is considered unstable.
 
 #### The run_data structure
 
-The `run_data` object contains all data from the Chef InSpec run. This object is a Hash, but includes many fields and is often large. No specific documentation exists for the `run_data` object. See [the legacy JSON reporter](https://github.com/inspec/inspec/blob/2e887a94afcca819da781d4774aa2a5a0b56785e/lib/inspec/reporters/json.rb#L10) for one example of how to iterate over the object.
+The `run_data` object contains all data from the Chef InSpec run. Here is an overview of this object:
+
+| Field | Description |
+|-------|-------------|
+|`run_data.control`| Array of Control records. Flattened copy of those found in |`run_data.profiles[*].controls[*]`. See there for details.|
```

Comment:
Thanks for taking the time to break out `run_data` in a way that can be documented and versioned as below, I find this really handy

Reply:
💯 You went far beyond what I expected here. Thanks.